HW14/optional-task-unit.py

```diff
@@ -0,0 +1,61 @@
+import unittest
+import functions as f
+import functions_with_errors as fe
+
+class TestStringMethods(unittest.TestCase):
+    def test_greeting_f(self):
+        print ("Testing greeting_by_name in functions.py")
+        data = ["Jon", "Bob", "ISHMAIL", "carrot1234", "cathulu the destroyer"]
+        for test in data:
+            with self.subTest(name=test):
+                self.assertEqual(f.greeting_by_name(test), f"Hello {test}!", "is Failed")
```

Comment:
The message 'is Failed' is not helpful because we already see this when a test fails. It would be better to explain the reason for failure - method greeting_by_name should return <expected> but returned <actual>